### DISCUSSION TITLE
Where is the real implementation for commands.entity(id)?

### DISCUSSION THREAD

**Discussion by DasLixou** [bevyengine/bevy#5926] [Q&A] [answered]
Where is the real implementation for commands.entity(id)?

I can only find that it resolves in an EntityCommand that then adds the commands to the top command when i am understanding it right, but where is the actual logic for that? Im kinda new to rust and dont really understand this kind of project organization :D thanks :D

**Comment by bjorn3** [comment, marked as answer]
This is the full implementation of `commands.entity(id)`: https://github.com/bevyengine/bevy/blob/7d9e864d9cbd2e755d2458b3d204328776687227/crates/bevy_ecs/src/system/commands/mod.rs#L249-L254 It calls `get_entity` which is defined at https://github.com/bevyengine/bevy/blob/7d9e864d9cbd2e755d2458b3d204328776687227/crates/bevy_ecs/src/system/commands/mod.rs#L283-L285 `EntityCommands` has a bunch of methods which add a command to the `commands` on which `.entity(id)` was originally called and use the passed in entity. For example `commands.entity(id).insert(my_component)` is implemented at https://github.com/bevyengine/bevy/blob/7d9e864d9cbd2e755d2458b3d204328776687227/crates/bevy_ecs/src/system/commands/mod.rs#L567-L571 and pushes a `Insert { entity: self.entity, component }` to the commands queue. The actual implementation of `Insert` that inserts the component can be found at https://github.com/bevyengine/bevy/blob/7d9e864d9cbd2e755d2458b3d204328776687227/crates/bevy_ecs/src/system/commands/mod.rs#L792-L803

**Reply by DasLixou** [reply]
So i have to look at world.get_entity 😅
